src/librustc_codegen_ssa/back/symbol_export.rs

```diff
@@ -288,24 +297,44 @@ fn upstream_monomorphizations_provider(
         cnum_stable_ids
     };
 
+    let drop_in_place_fn_def_id = tcx.lang_items().drop_in_place_fn();
+
     for &cnum in cnums.iter() {
         for (exported_symbol, _) in tcx.exported_symbols(cnum).iter() {
-            if let &ExportedSymbol::Generic(def_id, substs) = exported_symbol {
-                let substs_map = instances.entry(def_id).or_default();
-
-                match substs_map.entry(substs) {
-                    Occupied(mut e) => {
-                        // If there are multiple monomorphizations available,
-                        // we select one deterministically.
-                        let other_cnum = *e.get();
-                        if cnum_stable_ids[other_cnum] > cnum_stable_ids[cnum] {
-                            e.insert(cnum);
-                        }
+            let (def_id, substs) = match *exported_symbol {
+                ExportedSymbol::Generic(def_id, substs) => {
+                    (def_id, substs)
+                }
+                ExportedSymbol::DropGlue(ty) => {
+                    if let Some(drop_in_place_fn_def_id) = drop_in_place_fn_def_id {
+                        (drop_in_place_fn_def_id, tcx.intern_substs(&[ty.into()]))
+                    } else {
+                        // `drop_in_place` in place does not exist, don't try
+                        // to use it.
+                        continue
```

Comment:
Is there ever a case where `tcx.lang_items().drop_in_place_fn()` can validly return `None`?

Reply:
When `#[lang = "drop_in_place"]` is not defined.

Reply:
So for `#[no_core]` crates? In any case, the implementation here should be able to handle that gracefully.

Reply:
> So for #[no_core] crates?

Yes